app/src/src/screens/menu/MenuHomeScreen.jsx

```diff
@@ -1,104 +1,140 @@
+import { Card, IconButton, Paragraph } from 'react-native-paper';
 import MenuLogout from '../../components/buttons/MenuLogout';
 import { routes } from '../../utils/routes';
 import { getUserData } from '../../utils/storageAccess';
+import ProfilePhotoView from '../../components/views/ProfilePhotoView';
 
 // A screen that allows the user to access menu options
-const MenuHomeScreen = ({ navigation }) => {
-  const [user, setUser] = useState(null);
+class MenuHomeScreen extends Component {
+  state = { user: {} };
 
-  useEffect(() => {
-    (async () => {
+  constructor(props) {
+    super(props);
+  }
+
+  componentDidMount = async () => {
+    const userData = await getUserData();
+    this.setState({ user: userData });
+  };
+
+  componentDidUpdate = async (_prevProps, prevState) => {
+    if (prevState !== this.state) {
       const userData = await getUserData();
-      setUser(userData);
-    })();
-  }, []);
+      this.setState({ user: userData });
+    }
+  };
+
+  componentWillUnmount() {
+    this.setState = (_state, _callback) => {
+      return;
+    };
+  }
 
-  return user ? (
-    <View style={styles.container}>
-      <Header title="MenuHomeScreen" />
-      <View style={styles.backHeading}>
-        <Image
-          style={styles.backGroundHeading}
-          source={{ uri: user.background_photo }}
-        />
-        <View style={styles.profileArea}>
-          <TouchableOpacity
-            style={styles.profileImg}
-            onPress={() => navigation.navigate(routes.profile)}
-          >
-            <Avatar.Image
-              size={110}
-              style={styles.pfpShadow}
-              source={{ uri: user.profile_photo }}
+  render() {
+    return this.state.user ? (
+      <View style={styles.container}>
+        <Header title="MenuHomeScreen" />
+        <View style={styles.backHeading}>
+          {this.state.user.background_photo && (
+            <Image
+              style={styles.backGroundHeading}
+              source={{
+                uri: this.state.user.background_photo
+                  ? this.state.user.background_photo
+                  : '//:0',
+              }}
             />
-            <Text style={styles.profileName}>{user.preferred_name}</Text>
-            <Text style={styles.profileText}>See Your Profile</Text>
-          </TouchableOpacity>
+          )}
+          <View style={styles.profileArea}>
+            <TouchableOpacity
+              style={styles.profileImg}
+              onPress={() => this.props.navigation.navigate(routes.profile)}
+            >
+              <ProfilePhotoView
+                photo={this.state.user.profile_photo}
+                size={110}
+                styles={styles.pfpShadow}
+              />
+              <Text style={styles.profileName}>
+                {!!this.state.user.preferred_name
+                  ? this.state.user.preferred_name
+                  : ''}
```

Comment:
This can be simplified to 
`this.state.user.preferred_name ?? '//:0'`
This is called 'nullish coalescing', where the value on the right side of the ?? is used only if the left side is null or undefined. Otherwise, it will use the value on the left side.